src/header/internals/item.rs

```diff
@@ -82,6 +82,14 @@ impl Item {
         }
         self.typed.get_mut(tid).map(|typed| unsafe { typed.downcast_mut_unchecked() })
     }
+
+    pub fn into_typed<H: Header>(self) -> Option<H> {
```

Comment:
Ah yes, of course, because the `typed` is storing them as `Box<Header>`. I think there is a way to add a downcast taking self by value, such as something like `fn downcast(self: Box<Self>)`, but I'll toy on playground to find the right thing and report back.